modin/core/dataframe/algebra/binary.py

```diff
@@ -413,3 +413,47 @@ def caller(
                 )
 
         return caller
+
+    @classmethod
+    def apply(
+        cls, left, right, func, axis=0, func_args=None, func_kwargs=None, **kwargs
+    ):
+        r"""
+        Apply a binary function row-wise or column-wise.
+
+        Parameters
+        ----------
+        left : modin.pandas.DataFrame or modin.pandas.Series
+            Left operand.
+        right : modin.pandas.DataFrame or modin.pandas.Series
+            Right operand.
```

Comment:
This layer depends on the upper layer in every apply.

Reply:
The lower layer still takes the object(s) from the API layer, namely, modin.pandas.DataFrame/Series. Then, there is a `func`, which takes pandas.DataFrame/Series. Also, there is a kwargs argument that needs to be passed to the ``cls.register()``. What is ``cls.register`` for the user? Doesn't this look a little complicated for the user? So many things to understand. I am also thinking that we are trying to simplify the things not for the user, but for us ourselves, when we are re-writing a customer workload to get better performance.